build/Shared/NoAllocNewLineStreamWriter.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Text;
+
+namespace NuGet;
+
+internal class NoAllocNewLineStreamWriter : StreamWriter
```

Comment:
This class is only used by one project. Unless you expect that this will be used in other assemblies soon, I suggest moving it to that project. Also, I'd wrap the whole file in a `#if NETFRAMEWORK`, so that it doesn't bloat the size of assemblies that will never use it.

Reply:
Done!